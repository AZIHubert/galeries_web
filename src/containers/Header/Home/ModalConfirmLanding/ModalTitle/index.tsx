import * as React from 'react';

import Text from '#components/Text';

import {
  Container,
} from './styles';

const ModalTitle = () => (
  <Container>
    <Text
      styles={{
        fontSize: 1.1,
        textAlign: 'center',
      }}
      stylesMobile={{
        fontSize: 1.2,
      }}
      stylesLaptopL={{
        fontSize: 1.4,
      }}
    >
      Verify your email
    </Text>
  </Container>
);

export default ModalTitle;
