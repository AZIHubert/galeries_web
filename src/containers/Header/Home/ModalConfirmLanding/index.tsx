import * as React from 'react';
import {
  useDispatch,
  useSelector,
} from 'react-redux';

import Modal from '#components/Modal';
import Button from '#components/Button';
import Text from '#components/Text';

import {
  fetchSendConfirmation,
  resetSendConfirmation,
} from '#store/actions';
import { loadingSelector } from '#store/selectors';

interface ModalConfirmLandingI {
  currentEmail: string;
}

const ModalConfirmLanding = ({
  currentEmail,
}: ModalConfirmLandingI) => {
  const dispatch = useDispatch();
  const loading = useSelector(loadingSelector);

  React.useEffect(() => () => {
    dispatch(resetSendConfirmation());
  }, []);

  const onClick = () => {
    if (!loading) {
      dispatch(fetchSendConfirmation({
        email: currentEmail,
      }));
    }
  };

  return (
    <Modal.Container
      title={<Text
        textAlign='center'
      >
          Verify your email
      </Text>}
    >
      <p>
        To use Galeries, click the verification
        button in the email we sent
        to {currentEmail}. This helps keep
        your account secure.
      </p>
      {loading ? 'loading' : 'not loading'}
      <Button.Text
        disabled={loading}
        justifyContent='center'
        onClick={onClick}
        marginBottom={30}
        marginTop={20}
        text='No email in your inbox or spam folder? Letâ€™s'
        textButton='resend it'
      />
    </Modal.Container>
  );
};

export default ModalConfirmLanding;
