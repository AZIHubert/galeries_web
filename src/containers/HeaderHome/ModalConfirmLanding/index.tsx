import * as React from 'react';

import ModalContainer from '#components/ModalContainer';
import ModalTimer from '#components/ModalTimer';
import TextButton from '#components/TextButton';

import { LoadingContext } from '#contexts/LoadingContext';

import { resendConfirmation } from '#helpers/api';

interface ModalConfirmLandingI {
  currentEmail: string;
  setErrorModal: React.Dispatch<React.SetStateAction<{
    open: boolean;
    text: string;
  }>>;
}

const ModalConfirmLanding = ({
  currentEmail,
  setErrorModal,
}: ModalConfirmLandingI) => {
  const { loading, setLoading } = React.useContext(LoadingContext);
  const [open, setOpen] = React.useState<boolean>(false);
  const handleClose = () => setOpen(false);
  const onClick = async () => {
    setLoading(true);
    try {
      await resendConfirmation({ email: currentEmail });
      setOpen(true);
    } catch (err) {
      if (err.response) {
        if (err.status === 500) {
          setErrorModal({
            open: true,
            text: 'Something went wrong. Please try again',
          });
        } else {
          setErrorModal({
            open: true,
            text: err.response.data,
          });
        }
      } else {
        setErrorModal({
          open: true,
          text: 'Something went wrong. Please try again',
        });
      }
    }
    setLoading(false);
  };
  return (
    <ModalContainer
      testId="modalVerifyAccount"
      title='Verify your email'
      titleTextAlign='center'
    >
      <p
        data-testid='verifyAccountBody'
      >
        To use Galeries, click the verification
        button in the email we sent
        to {currentEmail}. This helps keep
        your account secure.
      </p>
      <TextButton
        disabled={loading}
        justifyContent='center'
        onClick={onClick}
        marginBottom={30}
        marginTop={20}
        text='No email in your inbox or spam folder? Letâ€™s'
        textButton='resend it'
      />
      <ModalTimer
        handleClose={handleClose}
        open={open}
        text='email resend'
      />
    </ModalContainer>
  );
};

export default ModalConfirmLanding;
