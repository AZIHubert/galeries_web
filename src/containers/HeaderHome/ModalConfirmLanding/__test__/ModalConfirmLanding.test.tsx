import * as React from 'react';
import renderer from 'react-test-renderer';

import ThemeProvider from '#contexts/ThemeContext';

import ModalConfirmLanding from '../index';

describe('ModalConfirmLanding', () => {
  it('renders without crashing', () => {
    const mockedSetErrorModal = jest.fn;
    const tree = renderer.create(
      <ThemeProvider>
        <ModalConfirmLanding
          setErrorModal={mockedSetErrorModal}
          currentEmail='mockedEmail@email.com'
        />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
