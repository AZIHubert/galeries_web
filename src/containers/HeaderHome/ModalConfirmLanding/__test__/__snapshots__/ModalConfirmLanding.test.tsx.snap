// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalConfirmLanding renders without crashing 1`] = `
<div
  className="sc-gsTCUz gpRCqf"
  data-testid="modalVerifyAccount"
>
  <div
    className="sc-dlfnbm erazeS"
  >
    <h3
      className="sc-hKgILt cfzZCa"
    >
      Verify your email
    </h3>
    <p
      data-testid="verifyAccountBody"
    >
      To use Galeries, click the verification button in the email we sent to 
      mockedEmail@email.com
      . This helps keep your account secure.
    </p>
    <div
      className="sc-gKsewC hQKVAW"
      fontSize={1}
    >
      <p>
        No email in your inbox or spam folder? Letâ€™s 
        <span
          className="sc-jSgupP eLLNoK"
          onClick={[Function]}
        >
          resend it
        </span>
        .
      </p>
    </div>
  </div>
  <div
    className="sc-bdfBwQ kUovzw background-container"
  />
</div>
`;
